### PR TITLE
Expression evaluation for deconstruction declaration

### PR DESCRIPTION
**Pull request by jcouv**
Allow a deconstruction declaration statement to be compiled by the expression evaluator.
Suggestions on additional EE tests are much appreciated, as it's my first time making a change there.

While debugging this and trying to dump the bound nodes, I ran into an issue with the `BoundTreeDumper` which prompted me to fix the names of some deconstruction fields with "Opt" suffix. 
The second commit unifies the nullability logic (one uses the Opt suffix, the other uses the `Null` xml attribute) in the code generator. It now only relies on the attribute. I checked that all fields with the "Opt" suffix have an appropriate "Null" attribute (both in C# and VB `BoundNodes.xml`).

Fixes https://github.com/dotnet/roslyn/issues/13265
@tmat @cston @dotnet/roslyn-compiler for review.

### CODE REVIEW

File: src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/ResultPropertiesTests.cs [hunk 1]
```diff
@@ -365,6 +365,48 @@ void Test()
         }
 
         [Fact]
+        public void DeconstructionDeclaration()
```
Comment (cston) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/ResultPropertiesTests.cs]: Please move test to `DeclarationTests` and verify the IL of the generated method. The IL should contain two calls to `IntrinsicMethods.CreateVariable`. See `DeclarationTests.Declarations` for an example.

**Comment by AlekseyTs** [comment]
LGTM, modulo the test strengthening.

File: src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/ResultPropertiesTests.cs (hunk 1 above)
Comment (jcouv) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/ResultPropertiesTests.cs]: Thanks for the tip. Added the IL verification.

**Comment by cston** [comment]
LGTM

**Comment by jcouv** [comment]
test windows_eta_open_prtest please

**Comment by jcouv** [comment]
test windows_eta_open_prtest please

**Pull request merged**